tempesta_fw/addr.c

```diff
@@ -44,48 +45,60 @@ validate_addr(const TfwAddr *addr)
 }
 
 static int
-tfw_inet_pton_ipv4(char **p, struct sockaddr_in *addr)
+tfw_inet_pton_ipv4(const char *pos, struct sockaddr_in *addr)
```

Comment:
Never rewrite code just because you don't understand it and can't properly adjust! I'm always happy to explain unclear pieces of code.

Moreover, the new implementation is poor since it examines the same data many times. Throw it out and fix the previous one.